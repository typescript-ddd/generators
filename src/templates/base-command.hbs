import { Command } from "@typescript-ddd/core";

/**
 * Represents a base class for {{natural entityName}} commands.
 {{#each properties}}
 * @param {{lbrace}}{{type}}{{rbrace}} {{name}} - The {{natural name}} of the {{natural ../entityName}}.
  {{/each}}
 */
export abstract class {{pascal entityName}}Command implements Command {
  protected constructor(
    {{#each properties}}
    public readonly {{name}}: {{type}},
    {{/each}}
  ) {
    // TODO: Add validation here
  }
}
