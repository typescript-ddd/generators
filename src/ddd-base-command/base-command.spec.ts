import { BaseCommandTemplateData } from "./base-command.types";
import { generateBaseCommand } from "./base-command.generator";

describe("Query", () => {

  it("should render template", () => {
    // Arrange
    const data: BaseCommandTemplateData = {
      entityName: "User",
      properties: [
        { name: "id", type: "UserId" },
      ],
    };

    // Act
    const result = generateBaseCommand(data);

    // Assert
    expect(result).toMatchSnapshot();
  });
});
