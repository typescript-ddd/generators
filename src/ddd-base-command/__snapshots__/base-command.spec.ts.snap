// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Query should render template 1`] = `
"import { Command } from "@typescript-ddd/core";

/**
 * Represents a base class for user commands.
 * @param {UserId} id - The id of the user.
 */
export abstract class UserCommand implements Command {
  protected constructor(
    public readonly id: UserId,
  ) {
    // TODO: Add validation here
  }
}
"
`;
